pkg/bundle/create/create.go

```diff
@@ -14,40 +15,53 @@ import (
 	"github.com/RedHatGov/bundle/pkg/bundle"
 	"github.com/RedHatGov/bundle/pkg/config"
 	"github.com/RedHatGov/bundle/pkg/config/v1alpha1"
+	"github.com/RedHatGov/bundle/pkg/metadata"
+	"github.com/RedHatGov/bundle/pkg/metadata/storage"
 	"github.com/RedHatGov/bundle/pkg/operator"
 )
 
+var (
+	// FullMetadataError should be returned when past runs exist in metadata.
+	FullMetadataError = errors.New("prior run metadata found, please run 'create diff' instead")
+
+	// FullMetadataError should be returned when no past runs exist in metadata.
+	DiffMetadataError = errors.New("no prior run metadata found, please run 'create full' instead")
+)
+
 // CreateFull performs all tasks in creating full imagesets
 func CreateFull(configPath, rootDir, outputDir string, dryRun, insecure bool) error {
 
 	ctx := context.Background()
 
-	sourceDir := filepath.Join(rootDir, config.SourcePath)
-	metadataPath := filepath.Join(sourceDir, "publish", config.MetadataFile)
+	sourceDir := filepath.Join(rootDir, config.SourceDir)
 
 	if err := bundle.MakeCreateDirs(rootDir); err != nil {
 		return err
 	}
 
-	// Read in current metadata
-	metadata, err := config.LoadMetadata(rootDir)
+	// TODO: make backend configurable.
+	backend, err := storage.NewLocalBackend(rootDir)
 	if err != nil {
-		return err
+		return fmt.Errorf("error opening local backend: %v", err)
 	}
 
-	// If metdata is found throw an error, else create new metadata
-	if len(metadata.PastMirrors) != 0 {
-		return config.NewFullMetadataError(metadataPath)
+	// Read in current metadata
+	meta := v1alpha1.Metadata{}
+	switch err := backend.ReadMetadata(ctx, &meta); {
+	case err != nil && !errors.Is(err, storage.ErrMetadataNotExist):
+		return err
+	case err == nil && len(meta.PastMirrors) != 0:
+		// No past run(s) are allowed when generating a full imageset.
+		// QUESTION: should the command just log a warning and ignore this metadata instead?
```

Comment:
@estroz Responding to your comment in the code, would it better to prompt the user or would we send a warning when creating a full and destroy the current metadata. @afflom and I talked about this and the user should be able to create a full even if there is existing metadata, but the only area of concern I have about this is the user accidentally running `create full`. Maybe a prompt or a metadata backup? WDYT?

Reply:
Prompt sounds fine. I'll create a follow-up.